File: leetcode/code/485.max_consecutive_ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        result = temp = 0
        nums.append(0)

        for i in nums:
            if i:
                temp += 1
            else:
                result = max(temp, result)
                temp = 0

        return result


if __name__ == "__main__":
    assert Solution().findMaxConsecutiveOnes([1, 1, 0, 1, 1, 1]) == 3

File: leetcode/code/485.max_consecutive_ones.go
package main

func findMaxConsecutiveOnes(nums []int) int {
    var result, temp = 0, 0
    nums = append(nums, 0)

    for _, num := range(nums) {
        if num == 1 {
            temp++
        } else {
            if temp > result {
                result = temp
            }
            temp = 0
        }
    }

    return result
}

func main() {
    array := []int{1, 1, 0, 1, 1, 1}
    if findMaxConsecutiveOnes(array) != 3 {
        println("buggy")
    }
}
